src/generators/jsx/utils/buildContent.mjs

```diff
@@ -0,0 +1,201 @@
+ *
+ * @param {string|undefined} sourceLink - The source link path
+ * @returns {import('hastscript').Element|null} The source link element or null if no source link
+ */
+const createSourceLink = sourceLink => {
+  if (!sourceLink) {
+    return null;
+  }
+
+  return createElement('span', [
+    'Source Code: ',
+    createElement(
+      'a',
+      { href: `${DOC_NODE_BLOB_BASE_URL}${sourceLink}` },
+      sourceLink
+    ),
+  ]);
+};
+
+/**
+ * Enhances a heading node with metadata, source links, and styling
+ *
+ * @param {ApiDocMetadataEntry} entry - The API metadata entry
+ * @param {import('mdast').Heading} node - The heading node to transform
+ * @param {number} index - The index of the node in its parent's children array
+ * @param {import('unist').Parent} parent - The parent node containing the heading
+ * @returns {[typeof SKIP]} Visitor instruction to skip the node
+ */
+const transformHeadingNode = (entry, node, index, parent) => {
+  const { data, children } = node;
+  const headerChildren = [
+    createElement(`h${data.depth + 1}`, [
+      createElement(`a.mark#${data.slug}`, { href: `#${data.slug}` }, children),
+    ]),
+  ];
+
+  // Add type icon if available
+  const iconSymbol = ICON_SYMBOL_MAP[data.type];
+  if (iconSymbol) {
+    headerChildren.unshift(createJSXElement('CircularIcon', iconSymbol));
+  }
+
+  // Add change history if available
+  const changeElement = createChangeElement(entry);
+  if (changeElement) {
+    headerChildren.push(changeElement);
+  }
+
+  // Replace node with new heading and anchor
+  parent.children[index] = createElement('div', headerChildren);
+
+  // Add source link if available
+  const sourceLink = createSourceLink(entry.source_link);
+  if (sourceLink) {
+    parent.children.splice(index + 1, 0, sourceLink);
+  }
+
+  return [SKIP];
+};
+
+/**
+ * Processes an API documentation entry by applying transformations to its content
+ *
+ * @param {ApiDocMetadataEntry} entry - The API metadata entry to process
+ * @returns {import('unist').Node} The processed content
+ */
+const processEntry = entry => {
+  // Create a copy to avoid modifying the original
+  const content = structuredClone(entry.content);
+
+  // Apply transformations
+  visit(content, createQueries.UNIST.isStabilityNode, transformStabilityNode);
+  visit(content, createQueries.UNIST.isHeading, (node, idx, parent) =>
+    transformHeadingNode(entry, node, idx, parent)
+  );
+
+  return content;
+};
+
+/**
+ * Creates the overall content structure with processed entries
+ *
+ * @param {Array<ApiDocMetadataEntry>} entries - API documentation metadata entries
+ * @param {Object} sideBarProps - Props for the sidebar component
+ * @param {Object} metaBarProps - Props for the meta bar component
+ * @returns {import('unist').Node} The root node of the content tree
+ */
+const createContentStructure = (entries, sideBarProps, metaBarProps) => {
+  return createTree('root', [
+    createJSXElement('NavBar'),
```

Comment:
Since all these are React Components, would make sense to _reference_ these as constants such as:

```
import { NavBar } from '@node-core/ui....'

...

createJSXElement(NavBar.displayName), // or something similar
```

This ensures that we use their real names; And then wherever we expose them (be it next.mdx.mjs and next.client.mdx.mjs) we also use the same constants.

This ensures that this won't randomly break if we chan the declared Components within MDX or somewhere else. Apply this to all components you're invoking within this new generator.

Reply:
So you import the constants, and then once they get IMPL you can update the constants file.

BTW, open an issue for this if you want to do it as a follow-up